syntax = "proto3";

package order;

service OrderService {
  rpc CreateOrder(CreateOrderRequest) returns (OrderResponse);
  rpc GetOrder(GetOrderRequest) returns (OrderResponse);
  rpc UpdateOrderStatus(UpdateOrderStatusRequest) returns (OrderResponse);
}

message CreateOrderRequest {
  string customer_name = 1;         // Add this field 
  string customer_email = 2;        // Add this field
  string customer_phone = 3;        // Add this field
  string restaurant_id = 4;
  repeated OrderItem items = 5;
  string delivery_address = 6;
  string special_instructions = 7;
}

message GetOrderRequest {
  string order_id = 1;
}

message UpdateOrderStatusRequest {
  string order_id = 1;
  OrderStatus status = 2;
}

message OrderItem {
  string item_id = 1;
  string name = 2;
  double price = 3;
  int32 quantity = 4;
  repeated string customizations = 5;
}

message OrderResponse {
  string order_id = 1;             
  string customer_name = 2;
  string customer_email = 3;
  string customer_phone = 4;
  string delivery_address = 5;
  string restaurant_id = 6;
  repeated OrderItem items = 7;
  string special_instructions = 8;
  OrderStatus status = 9;
  double total_amount = 10;
  string created_at = 11;
  string updated_at = 12;
  string estimated_delivery_time = 13;
}

enum OrderStatus {
  ORDER_UNKNOWN = 0;
  ORDER_PENDING = 1;
  ORDER_CONFIRMED = 2;
  ORDER_PREPARING = 3;
  ORDER_OUT_FOR_DELIVERY = 4;
  ORDER_DELIVERED = 5;
  ORDER_CANCELLED = 6;
}